### PR TITLE
upgrade to 0.38.0 with some customizations

### PR DESCRIPTION
**Pull request by gbates101**
This PR is insanely large but there are a few things to note:

- CI/CD has been fixed post-merge so it works with devspace and normal Jenkins deployments
- Permission syncing has been replaced with RPC calls (though projects and tables are still synced via rabbitmq)
- `charts/superset` is now defunct and has been replaced by `helm/superset`, which has been heavily modified to work in our ecosystem.

Files of note:
- `security.py` - Permission management logic has been replaced with RPC calls
- `auth_oidc.py` - Fixed hardcoded values and properly deletes session cookie on logout
- `event_handler.py` - No longer handles user events, only project and table events
- `_helpers.tpl` - Added Plaid role definition and configuration for RMQ, OIDC, etc
- `devspace.yaml` - Fixed local builds, added new config and custom commands for nodejs development

Reference: [ch5939](https://app.clubhouse.io/plaidcloud/story/5939/upgrade-superset-to-0-38)

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -67,6 +68,8 @@ packaging==20.4           # via bleach
 pandas==1.1.2             # via apache-superset
 parsedatetime==2.6        # via apache-superset
 pathlib2==2.3.5           # via apache-superset
+pika==1.1.0
+# -e git+git://github.com/PlaidCloud/plaid-rpc.git@master#egg=plaidcloud-rpc
```
Comment (rad-pat) [inline requirements/base.txt:72]: should this be commented out, or removed here?

File: superset/charts/filters.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 from typing import Any
 
 from flask_babel import lazy_gettext as _
-from sqlalchemy import or_
+from sqlalchemy import or_, and_
```
Comment (rad-pat) [inline superset/charts/filters.py:20]: Was `and_` used in the end?

**Review by rad-pat — COMMENTED** [review]
Minor comments

**Review by chucklay — APPROVED** [review]

File: superset/charts/filters.py (hunk 2 above)
Comment (gbates101) [reply superset/charts/filters.py:20]: Nope, doesn't look that way.

File: requirements/base.txt (hunk 1 above)
Comment (gbates101) [reply requirements/base.txt:72]: Yeah I'll remove it.

**Review by rad-pat — APPROVED** [review]

**Pull request merged**